File: app/src/main/java/intellij/class6/herancas/tarefa/tarefa01/java/Pessoa.java
package intellij.class6.herancas.tarefa.tarefa01.java;

public abstract class Pessoa {
    private String nome;
    private Data nascimento;

    public Pessoa(String nome, Data nascimento) {
        this.nome = nome;
        this.nascimento = nascimento;
    }

    public void imprimeDados() {
        System.out.println("Nome: " + nome + "\nNascimento: " + nascimento.getDia() + "/" + nascimento.getMes() + "/" + nascimento.getAno());
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Data getNascimento() {
        return nascimento;
    }

    public void setNascimento(Data nascimento) {
        this.nascimento = nascimento;
    }
}

File: app/src/main/java/intellij/class6/herancas/tarefa/tarefa01/kotlin/Pessoa.kt
package intellij.class6.herancas.tarefa.tarefa01.kotlin

abstract class Pessoa(var nome: String, var nascimento: Data) {

    open fun imprimeDados() {
        println(
            """
    Nome: $nome
    Nascimento: ${nascimento.dia}/${nascimento.mes}/${nascimento.ano}
    """.trimIndent()
        )
    }

}